react-hot-button/my-app/src/App.tsx

```diff
@@ -0,0 +1,18 @@
+import { useState } from 'react';
+import './App.css';
+import { ButtonContainer } from './ButtonContainer';
+
+function App() {
+  const [count, setCount] = useState(0);
+
+  function handleButtonClick() {
+    setCount(count + 1);
+  }
+  return (
+    <>
+      <ButtonContainer onButtonClick={handleButtonClick} count={count} />
+    </>
+  );
+}
```

Comment:
Totally valid. In this specific example, I would've had state and the event handler live inside of `ButtonContainer`. I understand it can be confusing considering the Q&A we just had, but you don't ALWAYS want the logic to live in the parent. It depends on the scenario.

Since we only ever render 1 instance of this component, and we want this component to behave the exact same way anywhere where it's used, it's better to keep the logic confined within the component itself. If you had plans to reuse this component in different ways, then it'd be better to move the logic up to the parent.